src/eddymotion/model/dmri_utils.py

```diff
@@ -0,0 +1,77 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright 2024 The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY kIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+import numpy as np
+
+DEFAULT_MIN_BVALUE = 0
+"""Minimum allowed value for the b-value."""
+
+DEFAULT_NUM_BINS = 15
+"""Number of bins to classify b-values."""
+
+DEFAULT_MULTISHELL_BIN_COUNT_THR = 7
+"""Default bin count to consider a multishell scheme."""
+
+
+def find_shelling_scheme(
+    bvals,
+    num_bins=DEFAULT_NUM_BINS,
+    multishell_nonempty_bin_count_thr=DEFAULT_MULTISHELL_BIN_COUNT_THR,
+):
+    """Find the shelling scheme on the given b-values: computes the histogram of
+    the b-values according to ``num_bins`` and depending on the nonempty bin
+    count, classify the shelling scheme as: single-shell if they are 2;
+    multi-shell if they are below the ``multishell_nonempty_bin_count_thr``
+    value; DSI otherwise.
+
+    Parameters
+    ----------
+    bvals : :obj:`~numpy.ndarray`
+         b-values in FSL format.
+    num_bins : :obj:`int`, optional
+        Number of bins.
+    multishell_nonempty_bin_count_thr : :obj:`int`, optional
+        Bin count to consider a multi-shell scheme.
+
+    Returns
+    -------
+    scheme : :obj:`str`
+        Shelling scheme.
+    hist : :obj:`~numpy.ndarray`
+        Histogram.
+    bin_edges : :obj:`~numpy.ndarray`
+        Bind edges.
+    """
+
+    bins = np.linspace(DEFAULT_MIN_BVALUE, max(bvals), num_bins + 1)
+    hist, bin_edges = np.histogram(bvals, bins=bins)
+
+    nonempty_bins = np.sum(hist > 0)
+
+    if nonempty_bins == 2:
+        scheme = "single-shell"
+    elif nonempty_bins < multishell_nonempty_bin_count_thr:
+        scheme = "multi-shell"
+    else:
+        scheme = "DSI"
+
+    return scheme, hist, bin_edges
```

Comment:
Should we just return `scheme` and the actual "shells"? For example, estimated as the median value of all the b-values in each nonempty bin.

Reply:
Reworked in b41a427 to return the actual values in each bin. Overall, not sure if the histogram approach is better or more robust than the k-means, but we'll find out as the tool is used. Resolving now. Ready to be merged.